src/windows-manager.cpp

```diff
@@ -887,14 +887,24 @@ namespace viewer {
                 itg->second->removeChild(n);
             }
             nodes_.erase (nodeName);
-	    // if node is of type collada, explicitely remove it from cache
-	    LeafNodeColladaPtr_t lnc(dynamic_pointer_cast<LeafNodeCollada>
-				     (n));
-	    if (lnc) lnc->removeFromCache();
             return true;
         }
     }
 
+    bool WindowsManager::removeObjectFromCache(const std::string& nodeName)
+    {
+        NodePtr_t node = getNode (nodeName, true);
+	// if node is of type collada, explicitely remove it from cache
+	LeafNodeColladaPtr_t lnc(dynamic_pointer_cast<LeafNodeCollada>(node));
+	if (!lnc){
+	  std::ostringstream os;
+	  os << "Node \"" << nodeName << "\" is node of type LeafNodeCollada.";
+	  throw std::invalid_argument(os.str().c_str());
+	}
+	lnc->removeFromCache();
```

Comment:
For consistency with the rest:
```suggestion
        FIND_NODE_OF_TYPE_OR_THROW (LeafNodeCollada, lnc, nodeName);
        lnc->removeFromCache();
```
Please, mind the spaces instead of tabs.